scene/3d/soft_body_3d.cpp

```diff
@@ -668,10 +675,10 @@ void SoftBody3D::pin_point_toggle(int p_point_index) {
 	pin_point(p_point_index, !(-1 != _has_pinned_point(p_point_index)));
 }
 
-void SoftBody3D::pin_point(int p_point_index, bool pin, const NodePath &p_spatial_attachment_path) {
+void SoftBody3D::pin_point(int p_point_index, bool pin, const NodePath &p_spatial_attachment_path, int p_insert_at) {
```

Comment:
This should validate that the `p_insert_at` range is valid, i.e. `>= -1` and `< pinned_points.size()`. Otherwise this will lead to hard-to-understand errors surfaced in CowData if users pass invalid values.

Reply:
I did think that would be the case, but I was unsure how CowData handled that so I was waiting for confirmation. It should be fixed now. Validation is done in the `_add_pinned_point()` method at line 753. Values out of bounds are simply added to the end of the list.

Reply:
My idea was more to abort and notify the user that they used the API illegally, with:
```
ERR_FAIL_COND_MSG(p_insert_at < -1 || p_insert_at >= pinned_points.size(), "Invalid index for pin point insertion position.");
```
at the top of this method (which is the end-user facing one).

Reply:
Ah yes, that makes more sense; I'll do that instead. I keep forgetting that's a user facing method.